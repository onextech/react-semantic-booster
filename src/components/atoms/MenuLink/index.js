import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { Menu } from 'semantic-ui-react';
import { HashLink } from 'react-router-hash-link';

const MenuLink = ({ to, children, ...props }) => {
  const isHashLink = to.charAt(0) === '#';

  const StyledLink = styled(isHashLink ? HashLink : Link)`
    width: 100%;
    display: flex;
    padding: .92857143em 1.14285714em;
    color: inherit;
    &:hover {
      color: inherit;
    }
    &:hover, &:focus {
      text-decoration: none;
    }
  `;

  return (
    <Menu.Item link fitted {...props} style={{ padding: 0 }}>
      <StyledLink to={to}>{children}</StyledLink>
    </Menu.Item>
  );
};

MenuLink.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.string,
    PropTypes.array,
  ]).isRequired,
  to: PropTypes.string.isRequired,
};

export default MenuLink;
