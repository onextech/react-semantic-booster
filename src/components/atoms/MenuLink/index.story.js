import React from 'react';
import { storiesOf } from '@storybook/react';
import { Menu } from 'semantic-ui-react';
import MenuLink from '.';

storiesOf('MenuLink', module)
  .add('Default', () => (
    <Menu>
      <MenuLink to="/">Home</MenuLink>
      <MenuLink to="#about">About</MenuLink>
      <MenuLink to="/google" link={{ target: '_blank' }}>External Link</MenuLink>
    </Menu>
  ));
