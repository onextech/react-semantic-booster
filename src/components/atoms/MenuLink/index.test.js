import React from 'react';
import renderer from 'react-test-renderer';
import { render } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import MenuLink from '.';


describe('<MenuLink>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Router><MenuLink to="/">Home</MenuLink></Router>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has a link child', () => {
    const wrapper = render(<Router><MenuLink to="/">Home</MenuLink></Router>);
    expect(wrapper.find('a').length).toBe(1);
  });
});
