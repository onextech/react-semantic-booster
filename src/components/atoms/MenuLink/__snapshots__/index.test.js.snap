// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink> renders correctly 1`] = `
<div
  className="link fitted item"
  onClick={[Function]}
  style={
    Object {
      "padding": 0,
    }
  }
>
  <a
    className="sc-bdVaJa gDaJBK"
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</div>
`;
